drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -1144,10 +1144,24 @@ void rswitch_enadis_rdev_irqs(struct rswitch_device *rdev, bool enable)
 void rswitch_trigger_chain(struct rswitch_private *priv,
 			   struct rswitch_gwca_chain *chain)
 {
-	if (!rswitch_is_front_priv(priv))
+	if (!rswitch_is_front_priv(priv)) {
 		rswitch_modify(priv->addr, GWTRC0, 0, BIT(chain->index));
-	else
-		rswitch_vmq_front_trigger_tx(chain->rdev);
+	} else {
+		struct rswitch_device *rdev = chain->rdev;
+
+		if (rdev->is_vmq) {
+			if (!READ_ONCE(rdev->vmq_info->scheduled_tx)) {
+				u64 back_rx, front_tx;
+
+				back_rx = READ_ONCE(rdev->vmq_info->back_rx);
+				front_tx = READ_ONCE(rdev->vmq_info->front_tx);
+				if (back_rx < front_tx)
+					rswitch_vmq_front_trigger_tx(rdev);
+			}
+		} else {
+			rswitch_vmq_front_trigger_tx(rdev);
+		}
+	}
```

Comment:
It's probably not the same, but check `gro_flush_timeout` and 'napi_defer_hard_irqs'. FYI

Reply:
Thank you for this info, I read about these fields in the net device structure. This is different scenario because these lines aimed not for repolling napi, but for retriggering HW from DomD side.